File: 0766-toeplitz-matrix/0766-toeplitz-matrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {
        
        int rows = matrix.length;
        int cols = matrix[0].length;

        for (int start_col = 0; start_col < cols; start_col++) {
            int val = matrix[0][start_col];
            int r = 1;
            int c = start_col + 1;
            while (r < rows && c < cols) {
                if (val != matrix[r][c]) {
                    return false;
                }
                r++;
                c++;
            }
        }

        for (int start_row = 1; start_row < rows; start_row++) {
            int val2 = matrix[start_row][0];
            int r = start_row + 1;
            int c = 1;
            while (r < rows && c < cols) {
                if (val2 != matrix[r][c]) {
                    return false;
                }
                r++;
                c++;
            }
        }
        return true;
    }
}

File: 0766-toeplitz-matrix/0766-toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        

        rows = len(matrix)
        cols = len(matrix[0])

        for start_col in range(cols):
            val = matrix[0][start_col]
            r, c = 1, start_col + 1
            while r < rows and c < cols:
                if val != matrix[r][c]:
                    return False
                r += 1
                c += 1
        
        for start_row in range(1, rows):
            val = matrix[start_row][0]
            r, c = start_row + 1, 1
            while r < rows and c < cols:
                if val != matrix[r][c]:
                    return False
                r += 1
                c += 1
        return True